x/streamer/types/distr_info.go

```diff
@@ -38,16 +39,19 @@ func (r DistrRecord) ValidateBasic() error {
 // DistrInfoFromDistribution converts sponsorship distribution to the DistrInfo type.
 // Returning an empty DistrInfo (with zero DistrInfo.TotalWeight) is a valid scenario.
 func DistrInfoFromDistribution(d sponsorshiptypes.Distribution) *DistrInfo {
+	totalWeight := math.ZeroInt()
+
 	records := make([]DistrRecord, 0, len(d.Gauges))
 	for _, g := range d.Gauges {
 		records = append(records, DistrRecord{
 			GaugeId: g.GaugeId,
 			Weight:  g.Power,
 		})
+		totalWeight = totalWeight.Add(g.Power)
 	}
 
 	return &DistrInfo{
-		TotalWeight: d.VotingPower,
+		TotalWeight: totalWeight,
 		Records:     records,
 	}
 }
```

Comment:
This is very subtle so IMO it would be a good idea to explain the precise intended behavior in more detail the docstring. Especially since it has changed couple times